libslide/src/grammar.rs

```diff
@@ -389,10 +429,47 @@ macro_rules! display_binary_expr {
 display_binary_expr!(<Expr>);
 display_binary_expr!(<ExprPat>);
 
-#[derive(PartialEq, Clone, Copy, Hash, Debug)]
+impl<E> PartialOrd for BinaryExpr<E>
+where
+    E: Expression,
+{
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl<E> Ord for BinaryExpr<E>
+where
+    E: Expression,
+{
+    fn cmp(&self, other: &Self) -> Ordering {
+        match self.op.cmp(&other.op) {
+            Ordering::Equal => match self.lhs.as_ref().cmp(&other.lhs.as_ref()) {
+                Ordering::Equal => self.rhs.as_ref().cmp(&other.rhs.as_ref()),
+                ord => ord,
+            },
```

Comment:
match within a match here is kind of ugly. can't it just be a single if statement. I know pattern matching is big in rust, but I think doubling up is odd.

Reply:
good point, done